### PR TITLE
Limit total size of all files uploaded

### PR DESCRIPTION
**Pull request by awead**
Enforces a limit when uploading files to a work. Total size of all files
uploaded via BrowseEverything and local files cannot exceed a configured
amount in bytes.

Monitors the total size with a progress bar:
exceeded
And shows an error when the limit is exceeded:
exceeded_2

### CODE REVIEW

File: app/assets/javascripts/sufia/save_work/uploaded_files.es6 [hunk 1]
```diff
@@ -22,12 +23,35 @@ export class UploadedFiles {
...
+        })
+        return total
+    }
+
+    get totalLimit() {
+        return this.element.data('limit')
+    }
+
+    get validateLimit() {
+        if (this.totalUploadSize < this.totalLimit) {
```
Comment (carolyncole) [inline app/assets/javascripts/sufia/save_work/uploaded_files.es6]: Should this be <=?

File: app/views/curation_concerns/base/_form_files.html.erb [hunk 2]
```diff
@@ -2,8 +2,9 @@
         make span an label
         Add aria-hidden to empty spans
         Put help text above the file list
+        Add total upload limit data element in bytes
 # %>
-<div id="fileupload">
+<div id="fileupload" data-limit="<%= Rails.application.config.upload_limit %>">
```
Comment (carolyncole) [inline app/views/curation_concerns/base/_form_files.html.erb:7]: 👍

File: config/application.rb [hunk 3]
```diff
@@ -35,6 +35,7 @@ class Application < Rails::Application
     config.doi_handle = ENV.fetch('doi_handle', 'doi:10.5072/FK2') # '10.18113/s1')
     config.doi_user = ENV.fetch('doi_user', '')
     config.doi_password = ENV.fetch('doi_password', '')
+    config.upload_limit = ENV.fetch('upload_limit', '1073741824')
```
Comment (carolyncole) [inline config/application.rb]: 10.gigabyte.to_s instead of '1073741824'

File: config/locales/en.yml [hunk 4]
```diff
@@ -75,6 +75,7 @@ en:
       available: "Files Available for Upload"
       caption: "Listing of files ready to be uploaded"
       local: "Files must be added before they can be uploaded"
+      limit_exceeded: "Selected files exceeds maximum allowed for upload"
```
Comment (carolyncole) [inline config/locales/en.yml]: can we add the limit into here?

**Review by carolyncole — COMMENTED** [review]
I know this is a WIP so I am guessing tests are coming.  I added a few inline comments, so our discussion from the standup does not get lost.

**Comment by awead** [comment]
@cam156 this is updated. It might need an accessibility review from @mtribone

**Review by carolyncole — APPROVED** [review]

**Comment by carolyncole** [comment]
@awead unfortunately there are conflicts

**Pull request merged**